src/Adapter/Carrier/Repository/CarrierRangeRepository.php

```diff
@@ -0,0 +1,213 @@
+        $results = $qb->executeQuery()->fetchAllAssociative();
+
+        return $results;
+    }
+
+    /**
+     * @param CarrierId $carrierId
+     * @param CarrierRangesCollectionInterface $rangesCollection
+     * @param ShopConstraint $shopConstraint
+     */
+    public function set(CarrierId $carrierId, CarrierRangesCollectionInterface $rangesCollection, ShopConstraint $shopConstraint): void
+    {
+        // Check shop constraint
+        $this->assertShopConstraint($shopConstraint);
+
+        // Get carrier
+        $rangeTable = $this->getRangeMethodTable($rangesCollection->getCalculatingMethod());
+
+        // Use transaction to ensure data consistency
+        $this->connection->beginTransaction();
+
+        // Reset carrier ranges
+        $this->reset($carrierId, $shopConstraint);
+
+        // Foreach zones
+        $rangesIds = [];
+        foreach ($rangesCollection->getZones() as $zone) {
+            $idZone = $zone->getZoneId();
+
+            // Foreach ranges in current zone
+            foreach ($zone->getRanges() as $range) {
+                // Check if range already exist in database
+                if (!in_array($range->getFrom() . '-' . $range->getTo(), array_keys($rangesIds), true)) {
+                    // Insert range
+                    $this->connection->insert(
+                        $this->dbPrefix . $rangeTable,
+                        [
+                            'id_carrier' => $carrierId->getValue(),
+                            'delimiter1' => $range->getFrom(),
+                            'delimiter2' => $range->getTo(),
+                        ]
+                    );
+                    $rangeId = $this->connection->lastInsertId();
+                    $rangesIds[$range->getFrom() . '-' . $range->getTo()] = $rangeId;
+                } else {
+                    $rangeId = $rangesIds[$range->getFrom() . '-' . $range->getTo()];
+                }
+
+                // Insert price in delivery table
+                $this->connection->insert(
+                    $this->dbPrefix . 'delivery',
+                    [
+                        'id_carrier' => $carrierId->getValue(),
+                        'id_' . $rangeTable => $rangeId,
+                        'id_zone' => $idZone,
+                        'price' => $range->getPrice(),
+                    ]
+                );
+            }
+        }
+
+        // Commit transaction
+        $this->connection->commit();
+    }
+
+    private function reset(CarrierId $carrierId, ShopConstraint $shopConstraint): void
+    {
+        $this->assertShopConstraint($shopConstraint);
+        $tablesToReset = [
+            'range_weight',
+            'range_price',
+            'delivery',
+        ];
+
+        foreach ($tablesToReset as $table) {
+            $this->connection->delete(
+                $this->dbPrefix . $table,
+                ['id_carrier' => $carrierId->getValue()]
```

Comment:
You need to add a `WHERE` condition here that latches the shopConstraint In our case, since we only handle all shops for now, it will always be `WHERE id_shop = NULL AND id_shop_group = NULL` but it's important to have this condition to avoid removing the data from previous versions that are assigned to a single shop or a group